### DISCUSSION TITLE
Difference in compilation flags

### DISCUSSION THREAD

**Discussion by demhademha** [PlummersSoftwareLLC/Primes#509] [Ideas]
Whilst looking through the project, I noticed that different programs are built with different flags. 
whilst PrimeCPP/solution_1 uses `-Ofast`, PrimeCPP/solution_4 uses `-O3`
The difference in compilation flags does affect the performance of a program. 
Ideally, each program should be built with the same flags (where appropriate of course)
Regards

**Comment by BlackMark29A** [comment]
I am generally not a fan of `-Ofast`, because it breaks standard compliance, which is why I chose `-O3`. But I don't think it actually matters in this case. One of the things `-Ofast` does is use `-ffast-math`, which allows the compiler to assume floating point operations are associative (and some other assumptions), which is not actually true, but also doesn't matter at all in this context, because there are no floating point values at all. The other thing `-Ofast` does is enable `-fallow-store-data-races`, which could potentially make a difference here, although I don't think there are any situations where the compiler could apply this optimization, but I might be wrong on that. I haven't actually tested it, but I will look into it when I have some time.

**Comment by rbergen** [comment]
From a project/maintainer perspective, we will not enforce uniformity concerning compiler flags, compilers, execution environments, etc. We acknowledge that they are relevant, but consider their selection and parameterization part of the solutions(' codebases) that are collected in this repo.

That said, this is a collaborative project, so if the contributors for a particular language decide to standardize on these points between themselves, then it is of course perfectly okay to open (a) PR(s) to effectuate this.

I'm going to convert this issue to a discussion of type Idea, so it can be discussed further, if desired.

**Comment by BlackMark29A** [comment]
I've now tested whether `-O3` vs `-Ofast` makes a difference and did not see any. I haven't tried many samples, but if there is a difference it is within the measurement error and so too small to make any appreciable difference.
